pkg/export/disk/disk.go

```diff
@@ -96,13 +131,17 @@ func (r *Writer) CloseConnection(connectionName string) error {
 	if !ok {
 		return fmt.Errorf("connection %s not found for disk driver", connectionName)
 	}
-	delete(r.openConnections, connectionName)
-	if conn.File != nil {
-		if err := conn.unlockAndCloseFile(); err != nil {
-			return fmt.Errorf("connection is closed without removing respective violations. error closing file: %w", err)
+	defer delete(r.openConnections, connectionName)
```

Comment:
here is an example of where there could be a race when multiple goroutines are trying to read/write to the same openConnections.

Reply:
caller function is taking care of locking - https://github.com/open-policy-agent/gatekeeper/blob/c8978ca9dbc19c3cd9f55de0ee5a2c656eba10bb/pkg/export/system.go#L72

Reply:
ack. IMO it's safer and more maintainable for the callee to maintain the lock internally since callers may forget to use locks.

Reply:
We implemented locking in System to make sure if a user is adding a custom driver on their own, then the System remains thread safe if locking is not implemented at driver level.